lib/BackbeatConsumer.js

```diff
@@ -265,8 +263,20 @@ class BackbeatConsumer extends EventEmitter {
     _tryConsume() {
         // use non-flowing mode of consumption to add some flow
         // control: explicit consumption of messages is required,
-        // needs explicit polling until new messages become available.
-        this._consumer.consume(this._concurrency, (err, entries) => {
+        // needs explicit polling to get new messages
+
+        const nNewConsumeRequests =
+              this._concurrency
+              - this._processingQueue.running()
+              - this._processingQueue.length()
+              - this._nConsumePendingRequests;
+        if (nNewConsumeRequests <= 0) {
+            // processing pipeline is 100% busy, retry later
+            return setTimeout(this._tryConsume.bind(this), 1000);
+        }
+        this._nConsumePendingRequests += nNewConsumeRequests;
+        return this._consumer.consume(nNewConsumeRequests, (err, entries) => {
+            this._nConsumePendingRequests -= nNewConsumeRequests;
```

Comment:
`this._nConsumePendingRequests -= nNewConsumeRequests;` this can be decremented only if there is no error fetching the entries correct?

Reply:
Actually I want to decrement it unconditionally, because this is the bookkeeper for pending *requests*, so even if the requests fail they are not pending anymore, and we don't want the counter to be off thereafter.